frontend/src/components/RelatedResources/TracePanel.tsx

```diff
@@ -6,11 +6,28 @@ import { TraceHeader } from '@/pages/Traces/TraceHeader'
 import { useTrace } from '@/pages/Traces/TraceProvider'
 import { TraceSpanAttributes } from '@/pages/Traces/TraceSpanAttributes'
 import { TraceVisualizer } from '@/pages/Traces/TraceVisualizer'
+import {
+	RelatedTrace,
+	useRelatedResource,
+} from '@/components/RelatedResources/hooks'
+import { useEffect, useState } from 'react'
 
-export const TracePanel: React.FC = () => {
+export const TracePanel: React.FC<{ resource: RelatedTrace }> = ({
+	resource,
+}) => {
 	const { highlightedSpan, loading, selectedSpan, traces } = useTrace()
 	const span = selectedSpan || highlightedSpan
 	const path = `${window.location.pathname}${window.location.search}`
+	const { updateQuery } = useRelatedResource()
+	const [query, setQuery] = useState(resource.query ?? '')
+
+	useEffect(() => {
+		setQuery(resource?.query ?? '')
+	}, [resource.query])
+
+	const handleSubmit = (query: string) => {
+		updateQuery(query)
```

Comment:
for logs, the behavior is to add to the query rather than to replace it. could we do the same here?
see https://www.loom.com/share/5aa4b73b6ad44686831eb31cee6fce7e

Reply:
Ack